contracts/base/Lenders.sol

```diff
@@ -64,28 +64,29 @@ contract Lenders is LendersInterface {
 
     /* Constructor */
 
-    constructor(
+    /** External Functions */
+
+    function initialize(
         address zTokenAddress,
         address lendingPoolAddress,
-        address interestConsensusAddress
-    ) public {
-        require(zTokenAddress != address(0x0), "zToken address is required.");
-        require(lendingPoolAddress != address(0x0), "LendingPool address is required.");
-        require(
-            interestConsensusAddress != address(0x0),
-            "Consensus address is required."
-        );
-        zToken = ZTokenInterface(zTokenAddress);
+        address interestConsensusAddress,
+        address settingAddress
+    ) external isNotInitialized() {
+        zTokenAddress.requireNotEmpty("ZTOKEN_MUST_BE_PROVIDED");
+        lendingPoolAddress.requireNotEmpty("LENDING_POOL_MUST_BE_PROVIDED");
+        interestConsensusAddress.requireNotEmpty("CONSENSUS_MUST_BE_PROVIDED");
+
+        initialize(settingAddress);
+
+        zToken = zTokenAddress;
         lendingPool = lendingPoolAddress;
         interestConsensus = InterestConsensusInterface(interestConsensusAddress);
     }
 
-    /** External Functions */
-
     function setAccruedInterest(
         ZeroCollateralCommon.InterestRequest calldata request,
         ZeroCollateralCommon.InterestResponse[] calldata responses
-    ) external {
+    ) external isInitialized() whenNotPaused() whenLendingPoolNotPaused() nonReentrant() {
```

Comment:
For what reason should this function be `whenLendingPoolNotPaused`? This function doesnt use the lending pool

Reply:
why is nonReentrant needed? It doesnt make any external function calls

Reply:
I think if the current lending pool is paused, we shouldn't allow executions on this function.